src/jdk.compiler/share/classes/com/sun/tools/javac/parser/DocCommentParser.java

```diff
@@ -1301,13 +1301,33 @@ public DCTree parse(int pos) throws ParseException {
                 }
             },
 
-            // {@inheritDoc}
+            // {@inheritDoc class-name}
             new TagParser(TagParser.Kind.INLINE, DCTree.Kind.INHERIT_DOC) {
                 @Override
                 public DCTree parse(int pos) throws ParseException {
+                    DCReference ref = reference(ReferenceParser.Mode.MEMBER_DISALLOWED);
+                    skipWhitespace();
                     if (ch == '}') {
                         nextChar();
-                        return m.at(pos).newInheritDocTree();
+                        // *** ATTENTION ***
+                        //
+                        // DocTreeFactory.newInheritDocTree() predates
+                        // DocTreeFactory.newInheritDocTree(ReferenceTree)
+                        //
+                        // Although it does not seem possible to plug in a custom
+                        // DocTreeFactory, for backward compatibility, we should
+                        // call the former method when there's no reference
+                        // rather the latter and pass null.
+                        //
+                        // Long story short: do NOT refactor the below if-else to
+                        //
+                        //     m.at(pos).newInheritDocTree(ref)
+                        //
```

Comment:
Although the scary warning is not wrong, I think it would have been enough to say something more concise, like "_use the original legacy method if no ref is given_"